ClientLib/src/main/java/org/droidplanner/services/android/impl/core/drone/autopilot/generic/GenericMavLinkDrone.java

```diff
@@ -237,13 +237,21 @@ public boolean isConnectionAlive() {
     }
 
     @Override
-    public byte getSysid() {
-        return heartbeat.getSysid();
+    public short getSysid() {
+
+        return getUnsignedShort(heartbeat.getSysid());
     }
 
     @Override
-    public byte getCompid() {
-        return heartbeat.getCompid();
+    public short getCompid() {
+        return getUnsignedShort(heartbeat.getCompid());
+    }
+
+    // [ToDo] Helper Method that should be refactored to utility class.
+    private short getUnsignedShort(byte aByte){
```

Comment:
@billbonney The `HeartBeat#sysid` and `HeartBeat#compid` field are coming in as `int` and being downcast to `byte`. Looks to me we should change them to short as well:
https://github.com/dronekit/dronekit-android/blob/0f1e038fb797839d5c98b041659b22ef2b2bd9bf/ClientLib/src/main/java/org/droidplanner/services/android/impl/core/drone/variables/HeartBeat.java#L67-L68